modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/policy/DefaultPolicyTemplateFactoryTestCase.java

```diff
@@ -76,6 +85,35 @@ public void createsPolicyTemplate() throws Exception {
     assertThat(regionClassLoader.getArtifactPluginClassLoaders().size(), equalTo(1));
   }
 
+  @Test
+  @Issue("MULE-19387")
+  public void maintainsPluginDependenciesOrder() throws Exception {
+    RegionClassLoader regionClassLoader = createRegionClassLoader();
+    PolicyTemplateClassLoaderBuilder policyTemplateClassLoaderBuilder = createPolicyTemplateClassLoaderBuilder(regionClassLoader);
+    MuleDeployableArtifactClassLoader policyClassLoader = mock(MuleDeployableArtifactClassLoader.class);
+
+    List<String> pluginNames =
+        Arrays.asList("plugin 1", "plugin 2", "plugin 3", "plugin 4", "plugin 5", "plugin 6", "plugin 7", "plugin 8", "plugin 9");
+    List<ArtifactPluginDescriptor> pluginDescriptors = new ArrayList<>(pluginNames.size());
+    List<ArtifactClassLoader> pluginArtifactClassLoaders = new ArrayList<>(pluginNames.size());
+    pluginNames.forEach(pluginName -> {
+      pluginDescriptors.add(new ArtifactPluginDescriptor(pluginName));
+      ArtifactClassLoader pluginArtifactClassLoader = mock(ArtifactClassLoader.class);
+      when(pluginArtifactClassLoader.getArtifactId()).thenReturn(pluginName);
+      pluginArtifactClassLoaders.add(pluginArtifactClassLoader);
+    });
+    when(pluginDependenciesResolver.resolve(anySet(), anyList(), anyBoolean())).thenReturn(pluginDescriptors);
+    when(policyClassLoader.getArtifactPluginClassLoaders()).thenReturn(pluginArtifactClassLoaders);
+
+    when(policyClassLoader.getArtifactId()).thenReturn(POLICY_ID);
+    when(policyTemplateClassLoaderBuilder.build()).thenReturn(policyClassLoader);
+    when(classLoaderBuilderFactory.createArtifactClassLoaderBuilder()).thenReturn(policyTemplateClassLoaderBuilder);
+
+    PolicyTemplate policyTemplate = policyTemplateFactory.createArtifact(createApplication(regionClassLoader), descriptor);
+
+    assertThat(policyTemplate.getOwnArtifactPlugins(), contains(policyTemplate.getArtifactPlugins().toArray()));
```

Comment:
use a hamcrest matcher that doesn't require converting to an array.

Reply:
Can't do. Using the list instead of the array generates a matcher that has no use here :S